File: ABC/401-450/402/d.cpp
#include<iostream>
#include<vector>
#include<algorithm>
#include<map>
using namespace std;

int main(){
  int n,m;
  cin >> n >> m;
  map<int,int> mp;
  for(int i=0;i<m;i++){
    int a,b;
    cin >> a >> b;
    mp[(a+b)%n]++;
  }
  long long ans = (long long)m*(m-1)/2;
  for(auto [val,cnt] : mp) ans -= (long long)cnt*(cnt-1)/2;
  cout << ans << endl;
  return 0;
}

File: ABC/401-450/402/d.py
def main():
  n,m = map(int,input().split())
  ans = m*(m-1)//2
  cnt = [0]*n
  for i in range(m):
    a,b = map(int,input().split())  
    cnt[(a+b)%n]+=1

  for i in range(n):
    ans -= cnt[i]*(cnt[i]-1)//2

  print(ans)
if __name__ == '__main__':
  main()


